plugins/outputs/cloudwatch/aggregator_test.go

```diff
@@ -136,19 +136,32 @@ func TestAggregator_ShutdownBehavior(t *testing.T) {
 	// verify the remaining metrics can be read after shutdown
 	// the metrics should be available immediately after the shutdown even before aggregation period
 	aggregationInterval := 2 * time.Second
-	tags := map[string]string{"d1key": "d1value", "d2key": "d2value", aggregationIntervalTagKey: aggregationInterval.String()}
+	tags := map[string]string{
+		"d1key":                   "d1value",
+		"d2key":                   "d2value",
+		aggregationIntervalTagKey: aggregationInterval.String()}
 	fields := map[string]interface{}{"value": 1}
 	timestamp := time.Now()
 	m := metric.New(metricName, tags, fields, timestamp)
 	aggregator.AddMetric(m)
-
-	//give some time to aggregation to do the work
-	time.Sleep(time.Second * 2)
-
+	// The Aggregator creates a new durationAggregator for each metric.
+	// And there is a delay when each new durationAggregator begins.
+	// So submit a metric and wait for the first aggregation to occur.
+	time.Sleep(aggregationInterval + time.Second)
```

Comment:
Okay, I think I see the problem. It's because we wait long enough to potentially have 2 aggregation intervals occur when we only want one? 

Intuitively, does that mean we want to sleep for aggregationInterval < x < 2 * aggregationInterval?

What I'm mildly concerned about is if someone in the future tweaks the interval above but doesn't update this and it becomes flakey again (if agg interval changes to 1 second, then the problem reoccurs). Should this sleep time be a magnitude of aggregation interval, like what happens on line 160?

Reply:
The point of this test case is to shutdown before aggregation has "naturally" occurred.
The problem is mainly because the `aggregating()` method has this line:
```
time.Sleep(now.Truncate(durationAgg.aggregationDuration).Add(durationAgg.aggregationDuration).Sub(now))
```
Whch means when a new metric name is first seen and a `durationAggregator` is started, it could take between `aggregationInterval` and `2*aggregationInterval` for the first Aggregation.

You would think the test shouldn't care, it can just shutdown right after `AddMetric` is called and expect the shutdown/flush to happen ASAP.

You'd be wrong because of the randomness of selecting when 2 cases are "ready":
- `case m := <-durationAgg.aggregationChan:`
- `case <-durationAgg.shutdownChan:`

After my change, the requirement is that:
1. the first sleep is between aggregationInterval and 2*aggregationInterval.
2. The next sleep... I just realized is not needed. I will remove it...

Reply:
This may be an edge case to address.
At runtime if CWA has a duration aggregation interval of 60 seconds.
It is possible that if a new metric name is first seen just before shutdown it will delay shutdown for up to `aggregationDuration`.